client/init.js

```diff
@@ -87,17 +88,45 @@ addEventListener('load', function() {
 
 	function startWatchingPosition() {
 		navigator.geolocation.watchPosition(function(position) {
-			try {
-				comms.send({
-					event: 'friend-updated',
-					position: position
-				})
-			} catch (e) {
-				console.error(e)
-			}
+			sendMessage({
+				event: 'friend-updated',
+				position: position
+			})
 		})
 	}
 
+	function setupHiddenEvent(){
+		var visibility = new Visibility()
+		function handleVisibilityChange(){
+			var hiding = document[visibility.hidden]
+			if (hiding){
+				timeoutId = window.setTimeout(function(){
+					sendMessage({
+						event: 'friend-updated',
```

Comment:
What's the timeout for? Also, I don't like the client being authoritative as to whether or not it's stale. That seems like a display concern.